### ISSUE TITLE
New occ functionality for the ransomware app 

### ISSUE THREAD

**Issue by mmattel** [owncloud/docs-server#805]
## WHAT Needs to be Documented?
References: https://github.com/owncloud/ransomware_protection/pull/311 (Use an external file for the blacklist)
New occ comands:
`occ ransomguard:blacklist:set-file`
`occ ransomguard:blacklist:update:from-file`
`occ ransomguard:blacklist:update:from-site`

## WHERE Does This Need To Be Documented (Link)?
https://doc.owncloud.com/server/10.11/admin_manual/configuration/server/occ_command.html#ransomware-protection-enterprise-edition-only

## WHY Should This Change Be Made?
New functionality

## (Optional) What Type Of Content Change Is This? 
- [x] New Content Addition
- [ ] Old Content Deprecation
- [ ] Existing Content Simplification
- [ ] Bug Fix to Existing Content

## (Optional) Which Manual Does This Relate To?
- [x] Admin Manual
- [ ] Developer Manual
- [ ] User Manual
- [ ] Android
- [ ] iOS
- [ ] Branded Clients
- [ ] Desktop Client
- [ ] Other

@jvillafanez @DeepDiver1975 I need:
* more info how this works, pls describe
* info on how the data structure of the file/webpage has to look like including examples
* default values/paths/filename if exist
* ...

@jnweiger fyi

**Comment by jvillafanez** [comment]
> more info how this works, pls describe

https://github.com/owncloud/ransomware_protection/pull/311 should contain information about how it's expected to behave.

Maybe the piece missing is that we expect the admin runs the `set-file` command first and then, periodically, either the `update:from-file` or `update:from-site` (likely the latest)


> info on how the data structure of the file/webpage has to look like including examples

For the file, the format is the same as the bundled one (https://github.com/owncloud/ransomware_protection/blob/master/blacklist.txt.dist).
For the site, the response must be a json matching the same format as https://fsrm.experiant.ca/api/v1/get


> default values/paths/filename if exist

For the `set-file`, there is no default you can set. I mean, you can set the location anywhere you want and there is no default value. It isn't possible to remove the set value, but you can change it so another location is used (you'll need to touch the DB to remove the value, so it isn't officially supported).
The `set-file` will copy the contents of the old file to the new one. If it's the first time, the old file will be the bundled one (blacklist.txt.dist); in other words, the blacklist.txt.dist file will be copied to the new location the first time.

The "update" command will operate over the file set using the `set-file`. If the location hasn't been set, it will use the "ransomware_blacklist.txt" in the ownCloud's data directory.

Basically:
* bundled "blacklist.txt.dist" file WON'T be modified (it might be updated when the app is updated because it's part of the app)
* "ransomware_blacklist.txt" file in the ownCloud's data directory will be used if the `set-file` command hasn't been run. It will be used as default location for the blacklist. `update:from-file` and `update:from-site` will write in that file updating the contents.
* a custom location location will be used if `set-file` has been used. Both update commands will use the location set.

**Comment by jnweiger** [comment]
https://doc.owncloud.com/server/next/admin_manual/enterprise/security/ransomware-protection/#file-extension-blacklist currently mentions
`Ransomware Protection ships with [a static extension list](https://fsrm.experiant.ca/) of more than 3,000 file extensions.`

The exact number with version 1.6.0 is 4798. suggest to update the wording to ... more than 4,500 ...

**Comment by jvillafanez** [comment]
Running the `set-file` a second time will change the location. The contents will be copied to the new location.

**Comment by pako81** [comment]
Ah ok, all fine then I guess. Thanks
